File: schemas/recipe-cooking/drizzle/shopping_lists.ts
// shopping_lists: Named grocery lists optionally linked to a meal plan.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { mealPlans } from "./meal_plans";
import { users } from "../../auth-rbac/drizzle/users";

export const shoppingLists = pgTable(
  "shopping_lists",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    mealPlanId: uuid("meal_plan_id").references(() => mealPlans.id, { onDelete: "set null" }),
    createdBy: uuid("created_by").notNull().references(() => users.id, { onDelete: "cascade" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_shopping_lists_created_by").on(table.createdBy),
    index("idx_shopping_lists_meal_plan_id").on(table.mealPlanId),
  ]
);

File: schemas/recipe-cooking/spacetimedb/shopping_lists.rs
// shopping_lists: User-created lists for purchasing ingredients.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = shopping_lists, public)]
pub struct ShoppingList {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,

    #[index(btree)]
    pub meal_plan_id: Option<String>, // FK → meal_plans.id (set_null)

    #[index(btree)]
    pub created_by: String, // FK → users.id

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
